lib/downloader.js

```diff
@@ -1,20 +1,45 @@
 'use strict';
 const ytdl = require('ytdl-core');
-const options = require('../config');
+const { youtubeDLConfig } = require('../config');
 const ffmpeg = require('fluent-ffmpeg');
+const { SponsorBlock } = require('sponsorblock-api');
+const sponsorBlock = new SponsorBlock(1234);
+
+const sponsorComplexFilter = async link => {
+  const videoID = (new URLSearchParams((new URL(link)).search)).get('v');
+  let segments = [];
+  let complexFilter = null;
+  try {
+    segments = await sponsorBlock.getSegments(videoID, 'sponsor');
```

Comment:
If this is for music videos, you might want to put all the categories here, or at least `music_offtopic` which is for non-music sections in music videos.

Reply:
ah good suggestion 👍 (great service btw @pavanjadhaw ) 

question while we have you are all the segments user entered or do you have some "ai/autoscraped" sections too.

i only ask as when it was first requested i was a bit skeptical that all videos would get hits, but when i tested i was impressed with every channel i could think of coming back with hits (though this could just be because they were still somewhat popular channels)

Reply:
No, it's all manual submissions, but as you noticed, coverage is extremely high, at least in English content released after the the extension started (mid 2019 for sponsor, mid 2020 for other categories).

Reply:
amazing